.github/workflows/codeql.yml

```diff
@@ -0,0 +1,116 @@
+name: "Golang Code Analysis"
+
+on:
+  push:
+    branches: [ "main", release-* ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  go-cache-name: go
+
+jobs:
+  analyze-on-linux:
+    name: Analyze on Linux
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
+        config-file: ./.github/codeql-config.yml
+
+    - name: Set up Go using version from go.mod
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: 'go.mod'
+
+    - uses: actions/cache@v3
+      with:
+        # In order:
+        # * Module download cache
+        # * Build cache (Linux)
+        # * Build cache (Mac)
+        path: |
+          ~/go/pkg/mod
+          ~/.cache/go-build
+          ~/Library/Caches/go-build
+          %LocalAppData%\go-build
+        key: ${{ runner.os }}-${{ env.go-cache-name }}-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ env.go-cache-name }}-
+
+    - name: "Build Application"
+      run: |
+        GOARCH=amd64 make bin
```

Comment:
why are we explicitly setting the arch?
could you add a comment?

Reply:
I removed the arch to let it detect automatically. Thanks.